yaml/step.go

```diff
@@ -74,6 +77,32 @@ func (s *StepSlice) UnmarshalYAML(unmarshal func(interface{}) error) error {
 		return err
 	}
 
+	// iterate through each element in the step slice
+	for _, step := range *stepSlice {
+		// implicitly set `pull` field if empty
+		if len(step.Pull) == 0 {
+			step.Pull = constants.PullNotPresent
+		}
+
+		// TODO: remove this in a future release
+		//
+		// handle true deprecated pull policy
+		//
+		// a `true` pull policy equates to `always`
+		if strings.EqualFold(step.Pull, "true") {
+			step.Pull = constants.PullAlways
+		}
```

Comment:
Set the deprecated `true` pull policy to `always` for the `github.com/go-vela/types/yaml.Step` type